### DISCUSSION TITLE
inconsistent behavior with same config from different location

### DISCUSSION THREAD

**Discussion by nabeelsaabna** [renovatebot/renovate#32596] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

Renovate has a group preset [groupsocketio](https://docs.renovatebot.com/presets-group/#groupsocketio) and it's part of the  `config:recommended`.

Running renovate CLI with:
1. `renovate_config.json`
```
{
  "endpoint": "https://api.github.com/",
  "platform": "github",
  "repositories": [ "ns-rnv-manual/config-order-test"],
  "requireConfig": "optional",
  "token": "ghp_****",

  "extends": [
    "config:recommended"
  ],

  "packageRules": [
    {
      "groupName": "my-custom-group",
      "matchDatasources": [ "npm" ]
    }
  ]
}
```
2. command

```
docker run 
 --rm -it 
 -e RENOVATE_CONFIG_FILE=/work/renovate_config.json  
 -v /home/nabeels/workspace/rnv-manual/renovate_config.json:/work/renovate_config.json 
 ghcr.io/renovatebot/renovate:39
```

results in my custom `packageRule` being first in the `Full resolved config and hostRules including presets` and therefore it becomes lowest priority and gets ignored (PR title is `Update socket.io packages`)
```
"packageRules": [
      {
        "groupName": "my-custom-group",
        "matchDatasources": [
          "npm"
        ]
      },
      {
        "matchPackageNames": [
          "*"
        ],
        "semanticCommitType": "chore"
      },
      {
        "matchDepTypes": [
          "dependencies",
          "require"
        ],
        "semanticCommitType": "fix"
      },
      ...
```

My expectations is that presets would be applied last (at least in the same config level?!)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by nabeelsaabna** [comment]
Ignore for now, I need to investigate more

**Reply by viceice** [reply]
https://docs.renovatebot.com/self-hosted-configuration/#globalextends

**Reply by nabeelsaabna** [reply]
Can we make that `ignorePresets` is also applied for `globalExtends` in case provided from the start ?
